data-pipeline/src/data_pipeline/datasets/clinvar.py

```diff
@@ -65,61 +69,71 @@ def _parse_submission(submission_element, trait_mapping_list_element):
                 mapping_element = trait_mapping_list_element.find(selector)
                 if mapping_element is not None:
                     break
 
+        # 2. Match on the 'Preferred' name of an element, when there's no XRefs defined
         if mapping_element is None:
             preferred_name_element = trait_element.find("./Name/ElementValue[@Type='Preferred']")
             if preferred_name_element is not None and trait_mapping_list_element is not None:
-                selector = f"./TraitMapping[@ClinicalAssertionID='{submission_element.attrib['ID']}'][@TraitType='{trait_element.attrib['Type']}'][@MappingType='Name'][@MappingValue=\"{preferred_name_element.text}\"]"  # noqa
+                selector = f"./TraitMapping[@ClinicalAssertionID='{submission_element.attrib['ID']}'][@TraitType='{trait_element.attrib['Type']}'][@MappingType='Name'][@MappingValue='{escape_xpath_value(preferred_name_element.text)}']"  # noqa
                 mapping_element = trait_mapping_list_element.find(selector)
 
+        # 3. Match on the name of a classification, when there's no preferred names
         if mapping_element is None:
             name_elements = trait_element.findall("./Name/ElementValue")
             for name_element in name_elements:
                 if preferred_name_element is None:
                     preferred_name_element = name_element
 
                 if trait_mapping_list_element is not None:
-                    selector = f"./TraitMapping[@ClinicalAssertionID='{submission_element.attrib['ID']}'][@TraitType='{trait_element.attrib['Type']}'][@MappingType='Name'][@MappingValue=\"{name_element.text}\"]"  # noqa
+                    selector = f"./TraitMapping[@ClinicalAssertionID='{submission_element.attrib['ID']}'][@TraitType='{trait_element.attrib['Type']}'][@MappingType='Name'][@MappingValue='{escape_xpath_value(name_element.text)}']"  # noqa
                     mapping_element = trait_mapping_list_element.find(selector)
                     if mapping_element:
                         break
 
+        # After matching Classifications to TraitMappings, iterate over mapping elements and add the MedGen IDs
         if mapping_element is not None:
             medgen_element = mapping_element.find("./MedGen")
             submission["conditions"].append(
                 {"name": medgen_element.attrib["Name"], "medgen_id": medgen_element.attrib["CUI"]}
             )
+
+        # 4. If there's no mappings that can be made confidently, add a single entry without a MedGen ID
         elif preferred_name_element is not None:
             submission["conditions"].append({"name": preferred_name_element.text, "medgen_id": None})
 
     return submission
 
 
-def _parse_variant(variant_element):
+def _parse_variant(variant_element, tqdm_pbar=None):
     variant = {}
 
-    if variant_element.find("./InterpretedRecord") is None:
+    if variant_element.find("./ClassifiedRecord") is None:
         raise SkipVariant
```

Comment:
So, recognizing that you didn't introduce this one, you generally don't want to use exceptions for control flow that can be done, as it were, through normal channels. Here rather than raise an exception in the handful of places it happens, I would `return null`, thus bailing out of the function early, and adjust the logic that calls `_parse_variant` to handle `null` results.